src/client.ts

```diff
@@ -736,7 +751,7 @@ export class StreamChat<
     // reset client state
     this.state = new ClientState();
     // reset token manager
-    this.tokenManager.reset();
+    setTimeout(this.tokenManager.reset); // delay reseting to use token for disconnect calls
```

Comment:
whats the reason for this `setTimeout`? Just curious what was the usecase for it

Reply:
basically it's there to change the order of execution without breaking change which is after closeConnection sync part is done(reading the token one last time) and this function returns

Reply:
maybe `return closePromise.then(this.tokenManager.reset);` would be better for guaranteeing the order of execution?